### ISSUE TITLE
weekly | JJ

### ISSUE THREAD

**Issue by vamseeachanta** [vamseeachanta/energydata#7]
Objective Track the progress summary in a single issue board.

Example Format:
WE 2024-04-26:
- API input data -> detailed info.

**Comment by JayachandraJangiti** [comment]
@vamseeachanta 

Good morning Sir,

I have mailed you the extracting code (with virtual machine).

UML diagram of our work is completed using plantuml

You can find it below : 
https://github.com/JayachandraJangiti/WEB_SCRAPPING/blob/Extracting_table/Extract_table_content.plantuml

**Comment by JayachandraJangiti** [comment]
@vamseeachanta 

Good morning Sir,

Here is the code :

https://github.com/JayachandraJangiti/WEB_SCRAPPING/blob/Extracting_table/Extract_table_content.py

Here is the setup file : 

https://github.com/JayachandraJangiti/WEB_SCRAPPING/blob/Extracting_table/requirements.txt

You can run the Extract_table_content code using this setup file (contains all packages used with their versions).

**Comment by JayachandraJangiti** [comment]
@vamseeachanta 

Good morning Sir,

I have changed code to take all inputs from configuration (.yml) file. It takes all inputs from the file given and run the program for each input.

Here is the code :

https://github.com/JayachandraJangiti/WEB_SCRAPPING/blob/Extracting_table/main.py

Here is the Output : 

1) First of all It asks for the path of inputs file (.yml file). We need to give the path of .yml file in which inputs were written .

![Screenshot (24)](https://github.com/vamseeachanta/energydata/assets/159880775/438e9afe-94af-4ede-b04a-221511104306)

I have given below file as input 

https://github.com/JayachandraJangiti/WEB_SCRAPPING/blob/Extracting_table/Inputs.yml

The inputs are 

![Screenshot (29)](https://github.com/vamseeachanta/energydata/assets/159880775/cdef9363-a584-48d8-bd5a-b15aa63c80d3)




2) Then It prints dataframes for each input you have given in input file.


![Screenshot (25)](https://github.com/vamseeachanta/energydata/assets/159880775/253f7e60-ae57-41f6-89fa-427358f0648d)


![Screenshot (26)](https://github.com/vamseeachanta/energydata/assets/159880775/8bd59a5d-c286-4f77-8bd1-66c22ac352a6)

![Screenshot (27)](https://github.com/vamseeachanta/energydata/assets/159880775/3981057e-e305-4e0f-b2f9-b76f34c19bad)

![Screenshot (28)](https://github.com/vamseeachanta/energydata/assets/159880775/1d144e8b-9352-408e-ac34-a0ffcdac3c5c)

**Comment by vamseeachanta** [comment]
@JayachandraJangiti , I have rewired the data into the repository. Any future work should happen here. 

Please see setup instructions below and make sure you can run the test file. 
https://github.com/vamseeachanta/energydata/blob/bseedata/dev_tools/dev_instructions.md

**Comment by vamseeachanta** [comment]
short-term tasks:
- run in energydata repo. In branch bseedata
- get more data (well properties etc.)

long-term tasks:
a/ Run selenium code in background (without opening browser in foreground)
b/ rewire selenium code to get data in a generic way. transfer code to assetutilities 
c/ function inside a class function is confusing and not generic. Rewire to generic utilities.

**Comment by vamseeachanta** [comment]
@JayachandraJangiti , please give weekly updates. Else, accounting for work (and payments) will be difficult.

Also, did you get the energydata repo going? Did you get the virtual environment going? Need more prompt communications from you.

**Comment by vamseeachanta** [comment]
self-installing library instructions are given in this link:
https://github.com/vamseeachanta/py_package

**Comment by JayachandraJangiti** [comment]
@vamseeachanta @saiachanta2728 

Good morning Sir,

I and Sai have discussed, solved some problems and have done the following :

1) Created and worked with virtual Environments.
2) Had some problems like(Setting Environment variables in the laptop etc.) with miniconda3 and resolved those.
3) Cloned and  Worked on both assetutilities and energydata repos.
4) Have gone to run .yml files, but we got stuck there (We got some errors).
